nvi-rest/src/test/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandlerDbTest.java

```diff
@@ -63,14 +68,74 @@ void shouldUpdateApprovalStatus(NviApprovalStatus status) throws IOException {
+        var institutionId = randomUri();
+        var dbCandidate = createCandidate(institutionId);
+        var candidate = nviCandidateRepository.create(dbCandidate,
+                                                      List.of(DbApprovalStatus.builder()
+                                                                  .institutionId(institutionId)
+                                                                  .status(DbStatus.PENDING)
+                                                                  .build()));
+
+        var req = new NviStatusRequest(candidate.identifier(), institutionId, NviApprovalStatus.REJECTED, "Denied!");
+        var request = createRequest(req, institutionId);
+        handler.handleRequest(request, output, context);
+        var gatewayResponse = GatewayResponse.fromOutputStream(output, CandidateResponse.class);
+        var bodyAsInstance = gatewayResponse.getBodyObject(CandidateResponse.class);
+        assertThat(bodyAsInstance.approvalStatuses().get(0).status(), is(equalTo(DbStatus.REJECTED)));
+        assertThat(bodyAsInstance.approvalStatuses().get(0).reason(), is(equalTo("Denied!")));
+    }
+
+    @Test
+    void shouldFailWhenRequireMessageIsMissing() throws IOException {
+        var institutionId = randomUri();
+        var dbCandidate = createCandidate(institutionId);
+        var candidate = nviCandidateRepository.create(dbCandidate,
+                                                      List.of(DbApprovalStatus.builder()
+                                                                  .institutionId(institutionId)
+                                                                  .status(DbStatus.PENDING)
+                                                                  .build()));
+
+        var req = new NviStatusRequest(candidate.identifier(), institutionId, NviApprovalStatus.REJECTED, null);
+        var request = createRequest(req, institutionId);
+        handler.handleRequest(request, output, context);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(Matchers.equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+    }
+
+    @Test
+    void shouldRemoveReasonWhenChangingAwayFromRejectedStatus() throws IOException {
+        var institutionId = randomUri();
+        var dbCandidate = createCandidate(institutionId);
+        var candidate = nviCandidateRepository.create(dbCandidate,
+                                                      List.of(DbApprovalStatus.builder()
+                                                                  .institutionId(institutionId)
+                                                                  .status(DbStatus.REJECTED)
+                                                                  .reason("I made a mistake")
+                                                                  .build()));
+
+        var req = new NviStatusRequest(candidate.identifier(),
```

Comment:
What about naming `req` variables as requestBody, think it as a little bit confusing to have two parameters which have almost the same names next to each other.